### PR TITLE
ADM-157; (Feat) Add optional children field for CRUD form

### PR DESCRIPTION
**Pull request by Slavik1999**
Add optional children field for CRUD form create and update. Implement this logic to Drawer;

### CODE REVIEW

File: admiral/crud/interfaces.ts [hunk 1]
```diff
@@ -25,6 +25,26 @@ export type CRUDActionsLocale = {
     paginationTotal: (total: number) => string
 }
 
+export type CRUDFormCreate =
+    | {
```
Comment (Slavik1999) [inline admiral/crud/interfaces.ts:29]: Это прекоммит сам автоформатирует так, ничего не могу поделать с этим

File: admiral/crud/interfaces.ts [hunk 2]
```diff
@@ -25,6 +25,26 @@ export type CRUDActionsLocale = {
...
+export type CRUDFormEdit =
+    | {
+          fields: React.ReactNode
+          children?: never
+      }
+    | {
+          fields?: never
+          children: React.ReactNode
+      }
+
```
Comment (arturvolokhin) [inline admiral/crud/interfaces.ts:47]: 1. You can combine two types into one and name it - "CRUDForm".
2. Why did you use the "never" type if JSX elements are always passed here?

File: admiral/crud/index.tsx [hunk 3]
```diff
@@ -173,15 +198,38 @@ function makeUpdatePage<RecordType>(config: CRUDConfig<RecordType>) {
...
-        if (!!config.form.create.fields && !config.update?.title) {
+        if (config.form.edit.fields && config.form.edit.children) {
+            console.error('Please provide "form.edit.fields" or "form.edit.children"')
+        }
+
+        if (!config.form.edit.fields && !config.form.edit.children) {
+            console.error('Please provide "form.edit.fields" or "form.edit.children"')
+        }
+
+        if (!!config.form.edit.fields && !config.update?.title) {
```
Comment (arturvolokhin) [inline admiral/crud/index.tsx:214]: (NIT): It isn't necessary to convert config.form.edit.fields to the boolean

File: admiral/crud/index.tsx [hunk 4]
```diff
@@ -133,7 +150,20 @@ function makeCreatePage<RecordType>(config: CRUDConfig<RecordType>) {
...
+                <>
+                    <Form.Fields>{fields}</Form.Fields>
+
+                    <Form.Footer>
+                        <BackButton basePath={path}>{actionsLocale.back}</BackButton>
+                        <Form.Submit>{actionsLocale.submit}</Form.Submit>
+                    </Form.Footer>
+                </>
+            )
+        }, [children, fields, path, actionsLocale])
```
Comment (arturvolokhin) [inline admiral/crud/index.tsx:166]: (NIT): Here's a way to make it easier:

```
const pageFormChildren = useMemo(
            () =>
                children ?? (
                    <>
                        <Form.Fields>{fields}</Form.Fields>

                        <Form.Footer>
                            <BackButton basePath={path}>{actionsLocale.back}</BackButton>
                            <Form.Submit>{actionsLocale.submit}</Form.Submit>
                        </Form.Footer>
                    </>
                ),
            [children, fields, path, actionsLocale],
        )
```

**Review by arturvolokhin — COMMENTED** [review]
Add comment as review

**Pull request merged**